pkg/api/object.go

```diff
@@ -221,11 +220,47 @@ func (h *Handler) Listbuckets(w http.ResponseWriter, r *http.Request) {
 	FormatJsonResponse(w, sortedBuckets)
 }
 
-func handleConnectionLookupFailure(err error, action string, w http.ResponseWriter) {
-	logger.Errorf("failed to get %s: %+v", action, err)
-	if util.IsEtcdKeyNotFound(err) {
-		w.WriteHeader(http.StatusNotFound)
-	} else {
+// GetBucket gets the bucket from the object store in this cluster.
+// GET
+// /objectstore/buckets/{USER_ID}
+func (h *Handler) GetBucket(w http.ResponseWriter, r *http.Request) {
+	bucketName := mux.Vars(r)["bucketName"]
+
+	user, rgwError, err := rgw.GetBucket(h.context, bucketName, h.config.ClusterHandler.GetClusterInfo)
+	if err != nil {
+		logger.Errorf("Error getting bucket (%s): %+v", bucketName, err)
+
+		if rgwError == rgw.RGWErrorNotFound {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	FormatJsonResponse(w, user)
+}
+
+// DeleteBucket deletes the bucket in the object store in this cluster.
+// DELETE
+// /objectstore/buckets/{BUCKET_NAME}
```

Comment:
does this operate in the context of a particular user? or can this delete anyone's buckets?